plugin/evm/vm.go

```diff
@@ -1166,10 +1171,10 @@ func attachEthService(handler *rpc.Server, apis []rpc.API, names []string) error
 	return nil
 }
 
-func (vm *VM) NewLimitOrderProcesser() LimitOrderProcesser {
+func (vm *VM) NewLimitOrderProcesser() (LimitOrderProcesser, error) {
 	var validatorPrivateKey string
 	var err error
-	if vm.config.IsValidator {
+	if vm.config.NodeType == "kitkat" {
```

Comment:
[doubt] have you purposely not used isMatcher function over here?

Reply:
good catch. i can't use `isMatcher` because `LimitOrderProcesser` is not yet initialized, but I have replicated the logic.